napari/utils/migrations.py

```diff
@@ -61,3 +62,49 @@ def _update_from_dict(*args, **kwargs):
         return _update_from_dict
 
     return _wrapper
+
+
+def add_deprecated_property(
+    obj: Any,
+    previous_name: str,
+    new_name: str,
+    version: str,
+    since_version: str,
+) -> None:
+    """
+    Adds deprecated property and links to new property name setter and getter.
+
+    Parameters
+    ----------
+    obj:
+        Class instances to add property
+    previous_name : str
+        Name of previous property, it methods must be removed.
+    new_name : str
+        Name of new property, must have its setter and getter implemented.
+    version : str
+        Version where deprecated property will be removed.
+    since_version : str
+        version when new property was added
+    """
+
+    if hasattr(obj, previous_name):
+        raise RuntimeError(f"{previous_name} attribute already exists.")
+
+    if not hasattr(obj, new_name):
+        raise RuntimeError(f"{new_name} property must exists.")
+
+    msg = trans._(
+        f"{previous_name} is deprecated since {since_version} and will be removed in {version}. Please use {new_name}",
```

Comment:
Just to disambiguate whose property this is.

```suggestion
        f"{obj}.{previous_name} is deprecated since {since_version} and will be removed in {version}. Please use {new_name}",
```